lib/ur/usage_report.go

```diff
@@ -422,23 +411,16 @@ func (s *Service) VerifyConfiguration(from, to config.Configuration) error {
 
 func (s *Service) CommitConfiguration(from, to config.Configuration) bool {
 	if from.Options.URAccepted != to.Options.URAccepted || from.Options.URUniqueID != to.Options.URUniqueID || from.Options.URURL != to.Options.URURL {
-		s.stopMut.RLock()
 		select {
 		case s.forceRun <- struct{}{}:
-		case <-s.stop:
+		default:
+			// s.forceRun is one buffered, so even though nothing
+			// was sent, a run will still happen after this point.
 		}
-		s.stopMut.RUnlock()
```

Comment:
There is a change in behaviour here. This used to block until the forceRun completed potentially, or atleast was accepted.
That might no longer be the case. I suspect it's ok, but suggest you verify if it really is.

Reply:
The relevant select above just has three options: stop, forceRun (resets timer) and timer fired (send a usage report). So the only way there could be a significant difference is if at the time of this being called, a report is being sent. Then this would have blocked until that is finished, now it immediately unblocks. The result is the same: As soon as the usage report goes through, another one happens. And on the side that changes the config I can't see any reason why someone should want to wait on the previous (not the scheduled) usage report being finished.